### PR TITLE
fix `static mut` warnings in `control-plane-agent` & `host-sp-comms`

### PR DESCRIPTION
**Pull request by hawkw**
Currently, the `control-plane-agent` task uses `static mut`s to declare
static buffers. These are guarded by an `AtomicBool` that ensures that
only a single mutable reference to these buffers is ever created.
However, they are still declared as `static mut`s and referenced by
creating an `&mut` reference to those `static mut`s, which now generates
a warning as it may be unsound in future Rust editions:

<details>

<summary>Compiler warnings:</summary>

```
warning: creating a mutable reference to mutable static is discouraged
     --> task/control-plane-agent/src/mgs_gimlet/host_phase2.rs:334:14
     |
334 |     unsafe { &mut PHASE2_BUF }
     |              ^^^^^^^^^^^^^^^ mutable reference to mutable static
     |
     = note: for more information, see issue #114447 <https://github.com/rust-lang/rust/issues/114447>
     = note: this will be a hard error in the 2024 edition
     = note: this mutable reference has lifetime `'static`, but if the static gets accessed (read or written) by any other means, or any other reference is created, then any further use of this mutable reference is Undefined Behavior
     = note: `#[warn(static_mut_refs)]` on by default
help: use `addr_of_mut!` instead to create a raw pointer
     |
334 |     unsafe { addr_of_mut!(PHASE2_BUF) }
     |              ~~~~~~~~~~~~~~~~~~~~~~~~

warning: creating a mutable reference to mutable static is discouraged
     --> task/control-plane-agent/src/mgs_gimlet.rs:1431:14
     |
1431 |     unsafe { &mut UART_TX_BUF }
     |              ^^^^^^^^^^^^^^^^ mutable reference to mutable static
     |
     = note: for more information, see issue #114447 <https://github.com/rust-lang/rust/issues/114447>
     = note: this will be a hard error in the 2024 edition
     = note: this mutable reference has lifetime `'static`, but if the static gets accessed (read or written) by any other means, or any other reference is created, then any further use of this mutable reference is Undefined Behavior
help: use `addr_of_mut!` instead to create a raw pointer
     |
1431 |     unsafe { addr_of_mut!(UART_TX_BUF) }
     |              ~~~~~~~~~~~~~~~~~~~~~~~~~

warning: creating a mutable reference to mutable static is discouraged
     --> task/control-plane-agent/src/mgs_gimlet.rs:1448:14
     |
1448 |     unsafe { &mut UART_RX_BUF }
     |              ^^^^^^^^^^^^^^^^ mutable reference to mutable static
     |
     = note: for more information, see issue #114447 <https://github.com/rust-lang/rust/issues/114447>
     = note: this will be a hard error in the 2024 edition
     = note: this mutable reference has lifetime `'static`, but if the static gets accessed (read or written) by any other means, or any other reference is created, then any further use of this mutable reference is Undefined Behavior
help: use `addr_of_mut!` instead to create a raw pointer
     |
1448 |     unsafe { addr_of_mut!(UART_RX_BUF) }
     |              ~~~~~~~~~~~~~~~~~~~~~~~~~

warning: creating a mutable reference to mutable static is discouraged
     --> task/control-plane-agent/src/mgs_gimlet.rs:1464:14
     |
1464 |     unsafe { &mut INSTALLINATOR_IMAGE_ID_BUF }
     |              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ mutable reference to mutable static
     |
     = note: for more information, see issue #114447 <https://github.com/rust-lang/rust/issues/114447>
     = note: this will be a hard error in the 2024 edition
     = note: this mutable reference has lifetime `'static`, but if the static gets accessed (read or written) by any other means, or any other reference is created, then any further use of this mutable reference is Undefined Behavior
help: use `addr_of_mut!` instead to create a raw pointer
     |
1464 |     unsafe { addr_of_mut!(INSTALLINATOR_IMAGE_ID_BUF) }
     |              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

warning: `task-control-plane-agent` (bin "task-control-plane-agent") generated 4 warnings
```

</details>

These warnings suggest that we use `addr_of_mut` instead, but rather
than doing that, I've wrapped these buffers in `UnsafeCell`s as part of
a new wrapper type for resource that can only be claimed once,
unimaginatively named `ClaimOnce`.

`UnsafeCell` expresses to the compiler that we intend to mutate this
data but allows us to store it in a regular (non `mut`) `static`
variable, which feels much less sketchy --- in general, the use of
`static mut` should be discouraged.

After making this change in `control-plane-agent`, I've also gone and
factored out the `ClaimOnce` type into the `static-cell` crate, so that it
can be used to remove one last use of `static mut` in `host-sp-comms`.

I haven't touched the `static mut` buffers in Hiffy, as I think that's a more
complex use-case that will require more thought.

### CODE REVIEW

File: task/attest/src/main.rs [hunk 1]
```diff
@@ -93,9 +92,8 @@ struct AttestServer {
 
 impl Default for AttestServer {
     fn default() -> Self {
```
Comment (cbiffle) [inline task/attest/src/main.rs:94]: This isn't new in your change, but a Default impl that can only ever be called once seems sketchy.

**Review by cbiffle — APPROVED** [review]
Looks generally good, the things I notice here aren't things you did.

File: task/thermal/src/control.rs [hunk 2]
```diff
@@ -524,11 +530,13 @@ impl<'a> ThermalControl<'a> {
...
-        use mutable_statics::mutable_statics;
-        let [err_blackbox, prev_err_blackbox] = mutable_statics! {
-            static mut ERR_BLACKBOX: [ThermalSensorErrors; 2] =
-                [Default::default; _];
-        };
+        use static_cell::ClaimOnceCell;
+
+        static ERR_BLACKBOX: ClaimOnceCell<ThermalSensorErrors> =
+            ClaimOnceCell::new(ThermalSensorErrors::new());
+        static PREV_ERR_BLACKBOX: ClaimOnceCell<ThermalSensorErrors> =
```
[the remark this answers is not in the document]
Comment (hawkw) [reply task/thermal/src/control.rs:537]: hm, they could be stuck in one `ClaimOnceCell` as a tuple or something, i guess...lemme do that.

File: task/attest/src/main.rs (hunk 1 above)
Comment (hawkw) [reply task/attest/src/main.rs:94]: yeah, i also noticed and didn't love that; I wasn't sure if it was worth changing as part of this branch or not, but i may go ahead and do it here.

**Comment by hawkw** [comment]
final size diff for `app/sidecar/rev-d.toml`:
```
Comparing against the previously saved sizes
Checking for differences in -caboose-
Checking for differences in auxflash
Checking for differences in control_plane_agent
	flash: -72
	ram: -4
Checking for differences in dump_agent
	flash: -88
	ram: 4
Checking for differences in ecp5_front_io
Checking for differences in ecp5_mainboard
Checking for differences in hiffy
Checking for differences in i2c_driver
Checking for differences in idle
Checking for differences in ignition
Checking for differences in jefe
Checking for differences in kernel
Checking for differences in monorail
Checking for differences in net
	flash: 2148
	ram: 4
Checking for differences in packrat
	flash: -36
Checking for differences in power
Checking for differences in sensor
Checking for differences in sequencer
Checking for differences in sprot
	flash: -40
	ram: 4
Checking for differences in sys
Checking for differences in thermal
	flash: -56
	ram: 8
Checking for differences in transceivers
	flash: -52
	ram: 4
Checking for differences in udpbroadcast
Checking for differences in udpecho
Checking for differences in update_server
Checking for differences in validate
Checking for differences in vpd
```

and for `app/gimlet/rev-f.toml`:
```
Comparing against the previously saved sizes
Checking for differences in -caboose-
Checking for differences in control_plane_agent
	flash: -240
Checking for differences in dump_agent
	flash: -88
	ram: 4
Checking for differences in gimlet_inspector
Checking for differences in gimlet_seq
Checking for differences in hash_driver
Checking for differences in hf
Checking for differences in hiffy
Checking for differences in host_sp_comms
	flash: -128
Checking for differences in i2c_driver
Checking for differences in idle
Checking for differences in jefe
Checking for differences in kernel
Checking for differences in net
Checking for differences in packrat
	flash: -52
Checking for differences in power
Checking for differences in sbrmi
Checking for differences in sensor
Checking for differences in spd
Checking for differences in spi2_driver
Checking for differences in sprot
	flash: -40
	ram: 4
Checking for differences in sys
Checking for differences in thermal
	flash: -68
	ram: 8
Checking for differences in udpbroadcast
Checking for differences in udpecho
Checking for differences in update_server
Checking for differences in user_leds
Checking for differences in validate
Checking for differences in vpd
```

File: task/thermal/src/control.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (cbiffle) [reply task/thermal/src/control.rs:537]: In the most recent stack of commits this looks like the only one that _didn't_ get put in a tuple :rofl:

**Pull request merged**